purchase_work_acceptance_invoice_plan/models/purchase.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    def action_view_wa(self):
+        res = super().action_view_wa()
+        return res
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    def _get_product_qty(self):
+        res = super()._get_product_qty()
+        installment_id = self._context.get("installment_id", False)
+        installment = self.env["purchase.invoice.plan"].browse(installment_id)
+        if installment_id:
+            for line in self:
+                res = line.product_qty * (installment.percent / 100)
+        return res
```

Comment:
```suggestion
        installment_id = self._context.get("installment_id", False)
        if installment_id:
            installment = self.env["purchase.invoice.plan"].browse(installment_id)
            return self.product_qty * (installment.percent / 100)
        return super()._get_product_qty()
```

Note:
* This method is called from 1 line always, and so, self = 1 line, no need to loop.
** https://github.com/OCA/purchase-workflow/blob/13.0/purchase_work_acceptance/models/purchase.py#L51
* We should run our logic before calling super. Otherwise, sometime it means unnecessary doing both.